optimizePNG.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+DIR_TO_MONITOR="/pixel/report"
+
+OPTIMIZATION_TAG="OptimizedWithOptiPNG"
+
+ENABLE_DEBUG_LOGGING=0
+
+debug_log() {
+  if [ "$ENABLE_DEBUG_LOGGING" -ne 1 ]; then
+    return
+  fi
+  echo "$(date +"%Y-%m-%d %H:%M:%S") - $1"
+}
+
+remove_lock_file() {
+  local LOCK_FILE
+  LOCK_FILE=$1
+  if [ -f "$LOCK_FILE" ]; then
+    rm -f "$LOCK_FILE" || true
+  fi
+}
+
+optimize_png() {
+  sleep 1
+
+  local FILE
+  FILE=$1
+
+  if exiftool -quiet -Software "$FILE" | grep -q "$OPTIMIZATION_TAG"; then
+    debug_log "$FILE already optimized, skipping"
+    return
+  fi
+
+  # Replace '/' with '_' in the file path to create a unique lock file name
+  local LOCK_FILE
+  LOCK_FILE="/tmp/$(echo "$FILE" | sed 's/\//_/g').lock"
+
+  if [ -f "$LOCK_FILE" ]; then
+    debug_log "$FILE already being processed, skipping"
+    return
+  fi
+
+  touch "$LOCK_FILE"
+
+  local SIZE_BEFORE
+  SIZE_BEFORE=$(stat -c%s "$FILE")
+
+  optipng -silent -fix -o2 "$FILE"
+
+  exiftool -quiet -overwrite_original_in_place -Software="$OPTIMIZATION_TAG" "$FILE"
```

Comment:
The lock file prevents this line from triggering another inotifywait create event

(exiftool is used to write "OptimizedWithOptiPNG" to the image's "Software" metadata so we have an easy way to see if a given image has already been optimized, as seen on line 30)